### DISCUSSION TITLE
Not usable for TYT MD-UV390

### DISCUSSION THREAD

**Discussion by githubrunbyidiotsthatforcecreateaccnts** [hmatuschek/qdmr#389] [Q&A]
Please help me. This application claims support for TYT MD-UV390 radios. I'm unable to use it at all.
I downloaded the latest release from https://github.com/hmatuschek/qdmr/releases/tag/v0.11.3 , this is a single QT app. It cannot open TYT codeplugs (.rdt format) nor DFU SPI flash dump. The button "read codeplug" does nothing, and honestly I do not see libusb/stdfu lubraries present in exe, so I can't comprehend how it can possibly read/write file from a device.

Creating a "new" codeplug does not allow selecting the radio model at all.

Please help me out? Does this thing work at all?

**Comment by allesand** [comment]
Some suggestions:

When asking for help, not calling other people idiots might be a good start?

Then, right at https://github.com/hmatuschek/qdmr "About" it says "A GUI application for configuring and programming cheap DMR radios under Linux and MacOS X."

Also, since qdmr supports a variety of radios, the process is to read from the radio to import a configuration, not to open whatever format the supplied software is using.

So, if the tool is not doing what one thinks it should do, there are several possible explanations:
- tool not suited for the specific task/wrong tool chosen
- tool not used in the way it was designed for/improper use of tool
- tool quality insufficient/quality issue

Assuming the last while not considering the former might lead to wrong conclusions.

**Reply by allesand** [reply]
I stand corrected: qdmr can read at least some manufacturers codeplugs: https://github.com/hmatuschek/qdmr/issues/64

**Comment by hmatuschek** [comment]
Also, consider RTFM: https://dm3mat.darc.de/qdmr/manual/

**Comment by tc0nn** [comment]
Although mine does have some "issues", I've been using Version 0.11.3 on my TYT md-uv390G. Are you downloading a precompiled version or building your own? Mac OSX Sonoma public beta.

**Reply by hmatuschek** [reply]
I think, @githubrunbyidiotsthatforcecreateaccnts used the inofficial Windows build.

**Reply by allesand** [reply]
@tc0nn What kind of issues are you experiencing? Since the RT3S and clones (or whichever way around..) are pretty widespread, there is a variety of firmware versions in use, what is your device running? Since listed on https://dm3mat.darc.de/qdmr/#dev they should work with qdmr.

**Comment by cpprioli** [comment]
I am having a possibly related issue.  Radio is an MD-UV390, base firmware S119.033 with OpenRTX 03.06 installed.  Qdmr for Windows cannot connect to the radio, returns an error message as shown.
![cap1371](https://github.com/user-attachments/assets/4c94a645-c7ea-4b0d-b830-22fff8dabc70)

Radio is in connection mode, and it shows up in Windows Device Manager as STM device in DFU mode.  What am I doing wrong?

**Comment by hmatuschek** [comment]
OpenRTX is not yet supported. However, there is some work in progress. See
issue #251.

cpprioli ***@***.***> schrieb am Mo., 10. Feb. 2025, 16:09:

**Comment by cpprioli** [comment]
Thanks for the quick replay.  I learned about qdmr on the OpenRTX site, where it is recommended.  I thought that it was being recommended for use with OpenRTX

**Comment by hmatuschek** [comment]
There were several delays on my side. The support was planned months ago. I
currently have basically no time to add new devices/firmwares to qdmr. So,
you need to be a bit more patient.

cpprioli ***@***.***> schrieb am Mo., 10. Feb. 2025, 16:16:

**Comment by cpprioli** [comment]
I will be very oatient.  I am in no hurry.  I can revert the radio to factory firmware...
